### DISCUSSION TITLE
How to best implement Zoom integration?

### DISCUSSION THREAD

**Discussion by mtdvlpr** [sircharlo/meeting-media-manager#1340] [Polls]
There is a Zoom integration in progress (see #1303 for details). There is an issue with using the video of the M3 Zoom participant. A decision needs to be made concerning how to implement this feature going forward. These are the option briefly explained (feel free to ask questions):

1. Don't use audio/video of the M3 participant. Can't auto connect audio, can mute and turn video off and can ask host to unmute and turn video on (Zoom popup)
2. Use host video and M3 audio. Spotlight on host required. Can ask host to turn video on (Zoom popup)
3. Support both options

Option 1 is sure to have the same audio/video quality as before, but the power of the integration is diminished.

Option 2 utilizes (almost) the full power, but the audio/video quality remains untested and forced spotlight adds complexity to the setup.

Option 3 allows users to choose and switch between the two options. If it turns out the audio quality is bad for option 2, option 1 can be selected. But it adds complexity to the feature and might confuse users on how to use this integration.

To be able to test all options, [the test version](https://drive.google.com/drive/folders/1_2CzKieYGaK3LcYrEwzNQgfTpNReOSx9?usp=sharing) has now implemented option 3 (both).

**Comment by sircharlo** [comment]
Sorry to resurrect an old thread here, but am I missing something? What is the "issue with using the video of the M³ Zoom participant"?
In my testing, the video of the M³ Zoom participant can be set to the OBS virtual camera, and the microphone can be set to whatever input is normally used by the normal Zoom client (to hear what goes on at KH).

If things are done that way, then the integration could remain a co-host, send itself to waiting room/main meeting as needed, etc., and there would be no need to spotlight the host, but rather "itself", since the AV feed would be coming from the M³ Zoom participant.

I might be missing something though like I said.

**Reply by mtdvlpr** [reply]
Yeah, so that would indeed be ideal, but when testing I and others found that the Zoom client claimed all available cameras, so you couldn't select any. I did not have this problem on Linux, but I did on Windows.

**Reply by sircharlo** [reply]
I just tested and didn't run into that; I wonder if something changed on their end.

**Reply by joaochris** [reply]
I didn't have that problem either. I can even use the OBS virtual camera in the integration and Zoom at the same time.

**Reply by mtdvlpr** [reply]
I'll test it again too!

**Reply by mtdvlpr** [reply]
Odd. I still have the same problem...I'm using Windows 10 with the latest Zoom client.

My steps:
1. Startup OBS with virtual camera
2. Startup Zoom Meeting and select any camera but the virtual camera
3. Startup M3 and open presentation mode
4. Select virtual camera in M3
5. See Zoom error
6. Select virtual camera in Zoom
7. See camera in action

**Reply by mtdvlpr** [reply]
Moved to #1642
